src/de.hpi.swa.trufflesqueak/src/de/hpi/swa/trufflesqueak/nodes/plugins/ffi/PrimExternalCallNode.java

```diff
@@ -0,0 +1,79 @@
+package de.hpi.swa.trufflesqueak.nodes.plugins.ffi;
+
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import de.hpi.swa.trufflesqueak.exceptions.PrimitiveFailed;
+import de.hpi.swa.trufflesqueak.nodes.primitives.AbstractPrimitiveNode;
+import de.hpi.swa.trufflesqueak.util.FrameAccess;
+import de.hpi.swa.trufflesqueak.util.NFIUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class PrimExternalCallNode extends AbstractPrimitiveNode {
+    final String moduleName;
+    final String functionName;
+    final int numReceiverAndArguments;
+    static Map<String, Object> loadedLibraries = new HashMap<>();
+
+    public PrimExternalCallNode(String moduleName, String functionName, int numReceiverAndArguments) {
+        this.moduleName = moduleName;
+        this.functionName = functionName;
+        this.numReceiverAndArguments = numReceiverAndArguments;
+    }
+
+    @Override
+    public Object execute(VirtualFrame frame) {
+        final Object uuidPlugin = loadedLibraries.computeIfAbsent(moduleName, (String s) -> NFIUtils.loadLibrary(getContext(), moduleName, "{ " +
```

Comment:
This should probably not be called uuidPlugin, right? Or is this specifically for the UUIDPlugin?